src/Microsoft.AspNet.Mvc.TagHelpers/JavaScriptHelpers.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+	internal static class JavaScriptHelpers
+    {
+        private static readonly Assembly ResourcesAssembly = typeof(JavaScriptHelpers).GetTypeInfo().Assembly;
+        private static readonly ConcurrentDictionary<Tuple<string, bool>, string> Cache = new ConcurrentDictionary<Tuple<string, bool>, string>();
+        
+        /// <summary>
+        /// Gets an embedded JavaScript file resource and optionally decodes it for use as a .NET format string.
+        /// </summary>
+    	public static string GetEmbeddedJavaScript(string resourceName, bool isFormatString = false)
```

Comment:
See my comment to @pranavkm, I'll remove it from the key, but the separation of the format string preparation isn't preemptive IMO.